core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1822,6 +1822,14 @@ private[spark] class BlockManager(
     }
   }
 
+  /*
+   *  Returns the last migration time and a boolean for if all blocks have been migrated.
+   *  If there are any tasks running since that time the boolean may be incorrect.
+   */
+  private[spark] def lastMigrationInfo(): (Long, Boolean) = {
```

Comment:
What do you think of directly exposing the "BlocksMigrated" and "BlocksToMigrate" instead of a single boolean ? I think it will provide more debugging value than a single flag. 

ie, instead of returning "BlocksMigrated >= BlocksToMigrate", return the two counters directly.

Or maybe this may be a lot more work because you want to track this for both shuffle as well as persisted blocks.

Reply:
I'd really rather not. I think the level above us should only be concerned if the blocks are done migrating or not. If we expose that up a level the logic is going to get even more complicated cross-class boundaries.

Reply:
Cool !. I am always anal about logging to a fault :-P